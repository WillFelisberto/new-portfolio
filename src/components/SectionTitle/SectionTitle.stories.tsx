import { Meta, StoryObj } from '@storybook/react';
import SectionTitle from '.';

const meta: Meta<typeof SectionTitle> = {
  title: 'Components/SectionTitle',
  component: SectionTitle,
  parameters: {
    layout: 'centered',
  },
  args: {
    title: 'Section Title',
    subtitle: 'Subtitle Example',
    className: '',
  },
};

export default meta;

type Story = StoryObj<typeof SectionTitle>;

export const Default: Story = {};

export const CustomClass: Story = {
  args: {
    className: 'text-center text-gray-500',
  },
};

export const LongText: Story = {
  args: {
    title: 'This is a much longer title to test how the component handles it',
    subtitle: 'A longer subtitle to test responsiveness and styles',
  },
};
