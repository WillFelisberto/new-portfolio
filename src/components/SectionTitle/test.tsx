/* eslint-disable @typescript-eslint/no-explicit-any */
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import SectionTitle from '.';

jest.mock('framer-motion', () => ({
  motion: {
    span: ({ children, ...props }: any) => <span {...props}>{children}</span>,
    h3: ({ children, ...props }: any) => <h3 {...props}>{children}</h3>,
  },
}));

describe('<SectionTitle />', () => {
  it('renders the title and subtitle with default styles', () => {
    const { container } = render(
      <SectionTitle title="Test Title" subtitle="Test Subtitle" />,
    );

    const subtitleElement = screen.getByText('../Test Subtitle');
    const titleElement = screen.getByText('Test Title');

    // Verifica se os textos estão na tela
    expect(subtitleElement).toBeInTheDocument();
    expect(titleElement).toBeInTheDocument();

    // Verifica classes padrão
    expect(subtitleElement).toHaveClass('font-mono text-sm text-sky-400');
    expect(titleElement).toHaveClass('text-3xl font-medium');

    expect(container).toMatchSnapshot();
  });

  it('applies additional className to the container', () => {
    render(
      <SectionTitle
        title="Test Title"
        subtitle="Test Subtitle"
        className="extra-class"
      />,
    );

    const container = screen.getByText('Test Title').closest('div');
    expect(container).toHaveClass('extra-class');
  });

  it('renders the motion props for animation', () => {
    render(<SectionTitle title="Test Title" subtitle="Test Subtitle" />);

    const subtitleElement = screen.getByText('../Test Subtitle');
    const titleElement = screen.getByText('Test Title');

    // Verifica se os atributos de animação estão aplicados
    expect(subtitleElement).toHaveAttribute('initial', expect.anything());
    expect(titleElement).toHaveAttribute('whileinview', expect.anything());
    expect(titleElement).toHaveAttribute('exit', expect.anything());
  });
});
