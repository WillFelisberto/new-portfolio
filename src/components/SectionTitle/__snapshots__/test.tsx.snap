// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionTitle /> renders the title and subtitle with default styles 1`] = `
<div>
  <div
    class="flex flex-col gap-4"
  >
    <span
      class="font-mono text-sm text-sky-400"
      exit="[object Object]"
      initial="[object Object]"
      transition="[object Object]"
      whileinview="[object Object]"
    >
      ../Test Subtitle
    </span>
    <h3
      class="text-3xl font-medium"
      exit="[object Object]"
      initial="[object Object]"
      transition="[object Object]"
      whileinview="[object Object]"
    >
      Test Title
    </h3>
  </div>
</div>
`;
